server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMDShardLockManager.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.apache.lucene.store.IndexOutput;
+
+import java.io.IOException;
+
+/**
+ * An Interface that defines Shard Level Remote Store MD Lock Manager.
+ * @opensearch.internal
+ */
+public interface RemoteStoreMDShardLockManager {
```

Comment:
What is the difference between `RemoteStoreMDShardLockManager` and `RemoteStoreMDIndexLockManager` other than naming?

Reply:
so major difference is Shard Lock will be having md File info and resourceID info, whereas Index Lock will have resourceId info and lock TTL.